src/instructlab/model/chat.py

```diff
@@ -839,6 +850,17 @@ def chat_cli(
     sys_prompt = CONTEXTS.get(context, "default")(get_model_arch(pathlib.Path(model)))
     loaded["messages"] = [{"role": "system", "content": sys_prompt}]
 
+    # Instantiate retriever if RAG is enabled
+    if ctx.obj.config.chat.rag.enabled:
+        logger.info("RAG enabled for chat; initializing retriever")
+        retriever: DocumentStoreRetriever | None = create_document_retriever(
```

Comment:
the factory function expects arguments of different type, this was the reason for my comment in the `configuration.py` file. At some point, we need to convert from one type to another, or have a single type either in `configuration` or `rag_configuration` module:
```py
def create_document_retriever(
    document_store_config: DocumentStoreConfig,
    retriever_config: RetrieverConfig,
) -> DocumentStoreRetriever:
```

Reply:
@dmartinol I hadn't looked through `rag_configuration.py`. It looks to me like that should be moved into `configuration.py` and follow the patterns there - I see that what I've done here along those lines overlaps with `rag_configuration.py` in PR this depends on.